pages/2_subscriptions_report.py

```diff
@@ -16,34 +17,113 @@
 ## Define data and filters. The resulting data variable includes the data with all filters applied.
 data = page_creation()
 
-st.divider()
-
-st.dataframe(data.head(10)) ## This is here just as an example. This can be deleted during development and before release.
-
-st.divider()
-
 # Everything below this point will be generated by the assigned analyst. 
 # The framework has been set, but no visualization or data processing has been applied.
 # Please perform any data processing in this file and not within the filter files. We can discuss upon completion if it makes sense to add any code to the filters file.
 
+# Convert payment_at to datetime if not already
+
+data.loc[:, 'payment_at'] = pd.to_datetime(data['payment_at'])
+data['payment_month'] = data['payment_at'].dt.to_period('M')
+data['subscription_started_month'] = data['subscription_period_started_at'].dt.to_period('M')
+
+# Filter the Dataframe to include only 'subscription' and 'recurring' billing types
+subscription_data = data[
+    (data['billing_type'].isin(['subscription', 'recurring'])) & 
+    (data['transaction_type'] == 'sale')
+]
+
+# Filter the Dataframe to include only 'one-time' and 'invoiceitem' billing types
+onetime_data = data[
+    (data['billing_type'].isin(['one-time', 'invoiceitem'])) & 
+    (data['transaction_type'] == 'sale')
+]
+
+# Group by 'payment_month' to calculate MRR and single order
+mrr_data = subscription_data.groupby(subscription_data['payment_month'])['total_amount'].sum().sort_index()
+single_order_data = onetime_data.groupby(onetime_data['payment_month'])['total_amount'].sum().sort_index()
+
 ## KPI Metrics which need to be updated.
 with st.container():
```

Comment:
For these metrics you are using Year over Year as the delta comparison. I normally would say this is appropriate and what we would realistically use with live data. However, this tells a confusing story with our sample data since we don't have that much data.

What are your thoughts around the following:
- Change these to be Quarter over Quarter to tell a bit more of an interesting story with the fake data we are using
- Adjust the fake data to include more data over a number of years so this can tell a more interesting story

Additionally, something to take into account here is that the `data` variable only includes the data provided in the filters. Therefore, we are not getting any data from anything outside of the min date in the date range. Therefore, if we want to do YoY then we will need to do the comparison with the raw data format that includes nearly everything. This is possible, but I think for arguments sake it would be easier to do QoQ and still rely on the `data` variable. What are your thoughts?